go/vt/sqlparser/sql.y

```diff
@@ -876,6 +923,10 @@ value_expression:
   {
     $$ = &CollateExpr{Expr: $1, Charset: $3}
   }
+| BINARY value_expression %prec UNARY
```

Comment:
I'd add tests to see how this plays with other unaries. Not sure how safe it is to overload that, but if it works than great.